iron-list.html

```diff
@@ -1870,15 +1872,31 @@
         }
       }
     },
-
-    _didMoveUp: function() {
-      this._focusPhysicalItem(this._focusedIndex - 1);
+    // When `grid === true` the `downArrow` goes forward `this._itemsPerRow` items
+    _didMoveBackward: function(e) {;
+      this._focusPhysicalItem(this._focusedIndex - (!this.grid ? 1 : this._itemsPerRow));
     },
-
-    _didMoveDown: function(e) {
+    // When `grid === true` the `upArrow` goes backward `this._itemsPerRow` items
+    _didMoveForward: function(e) {;
+      var destination = this._focusedIndex + (!this.grid ? 1 : this._itemsPerRow);
+	  // When `grid === true` move focus down one row, either
+	  // directly below or the last item of the list.
+	  if (this.grid && this._focusedIndex < (this.items.length - (this.items.length % this._itemsPerRow))) {
+		  destination = Math.min(destination, this.items.length - 1);
+	  }
```

Comment:
I don't have a strong preference for one behavior over the other, but if this is the desired result, you can just call `_focusPhysicalItem(this._focusedIndex + (this.grid ? this._itemsPerRow : 1))` with an out of range index and let the first `if` statement of `_focusPhysicalItem` do the early return. That way, this method would mirror the `_didMoveBackward` handler. Keep the test you wrote though to make sure this behavior doesn't unintentionally change in the future.